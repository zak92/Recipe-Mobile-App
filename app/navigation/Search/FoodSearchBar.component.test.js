
/**
 * @jest-environment jsdom
 */
import React from 'react';
import {render, fireEvent} from '@testing-library/react-native'; // json rendering
import FoodSearchBar from './FoodSearchBar.component';



describe('<FoodSearchBar />', () => {
  it("search page should match snapshot", () => {
    const snapFoodSearchBar = render(<FoodSearchBar onChangeText='hello' value='hello'/>).toJSON(); // render this file to JSON
    expect(snapFoodSearchBar).toMatchSnapshot(); // this snapshot should match the previous snapshot

  });
});


