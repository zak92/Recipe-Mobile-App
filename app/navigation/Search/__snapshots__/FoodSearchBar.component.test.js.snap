// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FoodSearchBar /> search page should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "transparent",
        "borderBottomWidth": 1,
        "borderTopColor": "transparent",
        "borderTopWidth": 1,
        "justifyContent": "center",
        "padding": "8%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderBottomWidth": 0,
            "borderColor": "grey",
            "borderRadius": 20,
            "color": "#000",
            "elevation": 5,
            "flexDirection": "row",
            "minHeight": 30,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Search food"
          placeholderTextColor="#86939e"
          platform="default"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "black",
              "flex": 1,
              "fontSize": 16,
              "marginLeft": 10,
              "minHeight": 40,
            }
          }
          testID="search"
          underlineColorAndroid="transparent"
          value="hello"
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginRight": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "height": 0,
            "margin": 0,
            "padding": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
